webapi/src/main/scala/org/knora/webapi/routing/v2/OntologiesRouteV2.scala

```diff
@@ -800,6 +817,53 @@ class OntologiesRouteV2(routeData: KnoraRouteData) extends KnoraRoute(routeData)
                                                            log = log
                                                          )
+
+              // get gui related values from request and validate them by making value objects from it
+
+              // get ontology info from request
+              inputOntology: InputOntologyV2             = InputOntologyV2.fromJsonLD(requestDoc)
+              propertyUpdateInfo: PropertyUpdateInfo     = OntologyUpdateHelper.getPropertyDef(inputOntology)
+              propertyInfoContent: PropertyInfoContentV2 = propertyUpdateInfo.propertyInfoContent
+
+              // get the (optional) gui element
+              maybeGuiElement: Option[SmartIri] =
+                propertyInfoContent.predicates
+                  .get(SalsahGui.External.GuiElementProp.toSmartIri)
+                  .map { predicateInfoV2: PredicateInfoV2 =>
+                    predicateInfoV2.objects.head match {
+                      case guiElement: SmartIriLiteralV2 => guiElement.value.toOntologySchema(InternalSchema)
+                      case other                         => throw BadRequestException(s"Unexpected object for salsah-gui:guiElement: $other")
+                    }
+                  }
+
+              // validate the gui element by creating value object
+              validatedGuiElement = maybeGuiElement match {
+                                      case Some(guiElement) => GuiElement.make(guiElement.toString()).map(Some(_))
+                                      case None             => Validation.succeed(None)
+                                    }
+
+              // get the gui attribute(s)
+              maybeGuiAttributes: List[String] =
+                propertyInfoContent.predicates
+                  .get(SalsahGui.External.GuiAttribute.toSmartIri)
+                  .map { predicateInfoV2: PredicateInfoV2 =>
+                    predicateInfoV2.objects.map {
+                      case guiAttribute: StringLiteralV2 => guiAttribute.value
+                      case other                         => throw BadRequestException(s"Unexpected object for salsah-gui:guiAttribute: $other")
+                    }.toList
+                  }
+                  .getOrElse(List())
+
+              // validate the gui attributes by creating value objects
+              guiAttributes = maybeGuiAttributes.map(guiAttribute => GuiAttribute.make(guiAttribute)).toList
+
+              validatedGuiAttributes = Validation.validateAll(guiAttributes)
+
+              // validate the combination of gui element and gui attribute by creating a GuiObject value object
+              guiObject: GuiObject = Validation
+                                       .validate(validatedGuiAttributes, validatedGuiElement)
+                                       .flatMap(values => GuiObject.make(values._1, values._2))
+                                       .fold(e => throw e.head, v => v)
```

Comment:
here again, we're dropping all but one error

Reply:
I will try the multiple errors handling in https://dasch.atlassian.net/browse/DEV-1082 (becuase right now, there is never more than one error)